### PR TITLE
Remove RHEL 7 from list of supported versions

### PR DESCRIPTION
**Pull request by sloretz**
There doesn't appear to be an active distro with it as a release platform.

### CODE REVIEW

**Comment by cottsay** [comment]
> There doesn't appear to be an active distro with it as a release platform.

I think this is true of all platforms listed here except for Ubuntu and RHEL 8.

**Comment by sloretz** [comment]
Ah hmm. I just assumed that supported mean there was an active ROS distro releasing into it. I'll close this.

Out of curiosity, what's the criteria for a platform to be in the supported list?

**Comment by cottsay** [comment]
> Out of curiosity, what's the criteria for a platform to be in the supported list?

I think there are two main criteria:
1. Upstream support - we don't check EOS/EOL platforms
2. Interest

The latter is a fairly fuzzy requirement that could probably be more rigidly defined somewhere.

The RHEL 7 case is an interesting one because it can actually make RHEL rule inclusion more complicated, where listing any single OS version for any of the other platforms will only make things easier. This pattern will probably pop up in other platforms that list multiple supported versions, but it's for more common in RHEL given how much the package set changes between releases.

tl;dr - there's probably a case for dropping RHEL 7. If we do that, I think we should do it completely, and not just stop checking rules against it.
